src/main/java/io/github/deweyjose/graphqlcodegen/FileReducer.java

```diff
@@ -0,0 +1,76 @@
+package io.github.deweyjose.graphqlcodegen;
+
+import java.io.File;
+import java.util.*;
+
+public class FileReducer {
+
+  /**
+   * We sort the input files by their paths to ensure a consistent order for processing.
+   *
+   * We maintain three sets:
+   *   validatedFiles to store the files that pass all checks,
+   *   encounteredDirectories to keep track of encountered directories
+   *   encounteredPaths to keep track of encountered paths.
+   *
+   * We iterate through the sorted files and perform the following checks:
+   *   Directory overlap: Skip files that are in directories already encountered.
+   *   Duplicate files: Skip files that are duplicates.
+   *   Ancestry in already added directories: Skip files that have ancestry in directories already encountered.
+   *
+   * Files that pass all checks are added to the validatedFiles set, and the corresponding
+   * directories and paths are updated in the encountered sets.
+   * @param files
+   */
+  public static Set<File> verifySchemaPaths(Collection<File> files) {
+    // Sort the input files to ensure consistent order
+    List<File> sortedFiles = new ArrayList<>(files);
+    sortedFiles.sort(Comparator.comparing(File::getPath));
+
+    Set<File> validatedFiles = new HashSet<>();
+    Set<File> encounteredDirectories = new HashSet<>();
+    Set<String> encounteredPaths = new HashSet<>();
+
+    for (File file : sortedFiles) {
+      String path = file.getPath();
+
+      // Check for directory overlap
+      File parent = file.getParentFile();
+      if (parent != null && encounteredDirectories.contains(parent)) {
+        throw new IllegalArgumentException(
+          String.format("Overlap detected: %s contains %s", parent.getPath(), file.getPath())
+        );
+      }
+
+      // Check for duplicate files
+      if (encounteredPaths.contains(path)) {
+        continue; // Skip this file, it's a duplicate
+      }
+
+      // Check for ancestry in already added directories
+      while (parent != null) {
+        if (encounteredDirectories.contains(parent)) {
+          throw new IllegalArgumentException(
```

Comment:
Same thought here. Can you avoid throwing errors instead?

Reply:
this catches the case where they configured `/dir` and `/dir/a.graphls`, its unclear what they want. All the graphqls files under /dir or just a.graphls.

Reply:
Right, which is why the warning would help debug something being included that shouldn’t have been due to misconfiguration.

Reply:
This error shows up in the plug-in output when maven runs

Reply:
And it'll stop maven execution and force developers to fix. Otherwise they run into an error later on in the build that's super confusing